artemis-server/src/main/java/org/apache/activemq/artemis/core/protocol/ProtocolHandler.java

```diff
@@ -243,6 +243,18 @@ protected void decode(ChannelHandlerContext ctx, ByteBuf in, List<Object> out) t
          ctx.flush();
       }
 
+      @Override
+      public void exceptionCaught(ChannelHandlerContext ctx, Throwable cause) {
+         // We definitely want to log the failure
+         ActiveMQServerLogger.LOGGER.failureDuringProtocolHandshake(ctx.channel().remoteAddress().toString(), cause);
+
+         // However, what else should be done here? I've laid out a few options...
+         ctx.fireChannelInactive();
+         ctx.close(); // this is done frequently in other implementations of exceptionCaught()
+         ctx.fireExceptionCaught(cause);
```

Comment:
If you are fully handling the exception caught case you don't need to forward this, but if there's another handler that is in the chain that you want to see the event then you could forward it.